src/GitLink/Linker.cs

```diff
@@ -41,22 +41,37 @@ public static bool Link(string pdbPath, LinkOptions options = default(LinkOption
             {
                 sourceFiles = pdb.GetFilesAndChecksums().Keys.ToList();
 
-                repositoryDirectory = GitDirFinder.TreeWalkForGitDir(Path.GetDirectoryName(sourceFiles.First()));
-                if (repositoryDirectory == null)
+                if (options.GitWorkingDirectory != null)
```

Comment:
It's more a matter of not wanting to allow "". Either folks specify a working directory, or they don't. To initialize to "" doesn't make sense. Using string.IsNullOrWhitespace would allow them to input "" or even "   " which doesn't make any sense. 
If `LinkOptions` were a class instead of a struct, I'd enforce in the ctor. If you want, I'll change it to a class.